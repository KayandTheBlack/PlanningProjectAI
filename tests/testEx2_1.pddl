(define (problem autogen)
  (:domain worldsEndAndBeyond)

  (:objects
    Origin cityA cityB cityC cityD cityE cityF cityG cityH cityI cityJ cityK cityL cityM cityN cityO - city
    hotelA_A hotelA_B hotelA_C hotelB_A hotelB_B hotelB_C hotelC_A hotelC_B hotelC_C hotelD_A hotelD_B hotelD_C hotelE_A hotelE_B hotelE_C hotelF_A hotelF_B hotelF_C hotelG_A hotelG_B hotelG_C hotelH_A hotelH_B hotelH_C hotelI_A hotelI_B hotelI_C hotelJ_A hotelJ_B hotelJ_C hotelK_A hotelK_B hotelK_C hotelL_A hotelL_B hotelL_C hotelM_A hotelM_B hotelM_C hotelN_A hotelN_B hotelN_C hotelO_A hotelO_B hotelO_C - hotel
  )

  (:init
    (currentLocation Origin)

    (hotelAt hotelA_A cityA)
    (hotelAt hotelA_B cityA)
    (hotelAt hotelA_C cityA)
    (hotelAt hotelB_A cityB)
    (hotelAt hotelB_B cityB)
    (hotelAt hotelB_C cityB)
    (hotelAt hotelC_A cityC)
    (hotelAt hotelC_B cityC)
    (hotelAt hotelC_C cityC)
    (hotelAt hotelD_A cityD)
    (hotelAt hotelD_B cityD)
    (hotelAt hotelD_C cityD)
    (hotelAt hotelE_A cityE)
    (hotelAt hotelE_B cityE)
    (hotelAt hotelE_C cityE)
    (hotelAt hotelF_A cityF)
    (hotelAt hotelF_B cityF)
    (hotelAt hotelF_C cityF)
    (hotelAt hotelG_A cityG)
    (hotelAt hotelG_B cityG)
    (hotelAt hotelG_C cityG)
    (hotelAt hotelH_A cityH)
    (hotelAt hotelH_B cityH)
    (hotelAt hotelH_C cityH)
    (hotelAt hotelI_A cityI)
    (hotelAt hotelI_B cityI)
    (hotelAt hotelI_C cityI)
    (hotelAt hotelJ_A cityJ)
    (hotelAt hotelJ_B cityJ)
    (hotelAt hotelJ_C cityJ)
    (hotelAt hotelK_A cityK)
    (hotelAt hotelK_B cityK)
    (hotelAt hotelK_C cityK)
    (hotelAt hotelL_A cityL)
    (hotelAt hotelL_B cityL)
    (hotelAt hotelL_C cityL)
    (hotelAt hotelM_A cityM)
    (hotelAt hotelM_B cityM)
    (hotelAt hotelM_C cityM)
    (hotelAt hotelN_A cityN)
    (hotelAt hotelN_B cityN)
    (hotelAt hotelN_C cityN)
    (hotelAt hotelO_A cityO)
    (hotelAt hotelO_B cityO)
    (hotelAt hotelO_C cityO)

    (flight Origin cityA)
    (flight Origin cityB)
    (flight Origin cityC)
    (flight Origin cityD)
    (flight Origin cityE)
    (flight Origin cityF)
    (flight Origin cityG)
    (flight Origin cityH)
    (flight Origin cityI)
    (flight Origin cityJ)
    (flight Origin cityK)
    (flight Origin cityL)
    (flight Origin cityM)
    (flight Origin cityN)
    (flight Origin cityO)
    (flight cityA cityC)
    (flight cityA cityF)
    (flight cityA cityK)
    (flight cityB cityC)
    (flight cityB cityE)
    (flight cityB cityN)
    (flight cityC cityE)
    (flight cityC cityF)
    (flight cityC cityG)
    (flight cityD cityG)
    (flight cityD cityH)
    (flight cityD cityO)
    (flight cityE cityD)
    (flight cityE cityI)
    (flight cityE cityO)
    (flight cityF cityL)
    (flight cityF cityM)
    (flight cityF cityN)
    (flight cityG cityD)
    (flight cityG cityF)
    (flight cityG cityL)
    (flight cityH cityD)
    (flight cityH cityG)
    (flight cityH cityN)
    (flight cityI cityA)
    (flight cityI cityG)
    (flight cityI cityM)
    (flight cityJ cityC)
    (flight cityJ cityD)
    (flight cityJ cityL)
    (flight cityK cityB)
    (flight cityK cityE)
    (flight cityK cityI)
    (flight cityL cityG)
    (flight cityL cityI)
    (flight cityL cityK)
    (flight cityM cityB)
    (flight cityM cityE)
    (flight cityM cityH)
    (flight cityN cityF)
    (flight cityN cityH)
    (flight cityN cityL)
    (flight cityO cityH)
    (flight cityO cityJ)
    (flight cityO cityN)

    (= (citiesVisited) 0)
    (= (totalDays) 0)
    (= (minDaysInCity) 2)
    (= (maxDaysInCity) 4)

    (= (stayDuration Origin) 0)
    (= (stayDuration cityA) 0)
    (= (stayDuration cityB) 0)
    (= (stayDuration cityC) 0)
    (= (stayDuration cityD) 0)
    (= (stayDuration cityE) 0)
    (= (stayDuration cityF) 0)
    (= (stayDuration cityG) 0)
    (= (stayDuration cityH) 0)
    (= (stayDuration cityI) 0)
    (= (stayDuration cityJ) 0)
    (= (stayDuration cityK) 0)
    (= (stayDuration cityL) 0)
    (= (stayDuration cityM) 0)
    (= (stayDuration cityN) 0)
    (= (stayDuration cityO) 0)
    (= (cityInterest Origin) 0)
    (= (cityInterest cityA) 2)
    (= (cityInterest cityB) 2)
    (= (cityInterest cityC) 2)
    (= (cityInterest cityD) 3)
    (= (cityInterest cityE) 1)
    (= (cityInterest cityF) 2)
    (= (cityInterest cityG) 3)
    (= (cityInterest cityH) 3)
    (= (cityInterest cityI) 3)
    (= (cityInterest cityJ) 3)
    (= (cityInterest cityK) 2)
    (= (cityInterest cityL) 3)
    (= (cityInterest cityM) 2)
    (= (cityInterest cityN) 1)
    (= (cityInterest cityO) 2)
    (= (totalInterest) 0)

  )

  (:goal
    (and
      (>= (citiesVisited) 12)
      (>= (totalDays) 20)
    )
  )

  (:metric 
    minimize (totalInterest)
  )

)
