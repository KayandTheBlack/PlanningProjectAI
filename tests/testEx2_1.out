
ff: parsing domain file
domain 'WORLDSENDANDBEYOND' defined
 ... done.
ff: parsing problem file
problem 'AUTOGEN' defined
 ... done.


metric established (normalized to minimize): ((1.00*[RF0](TOTALINTEREST)) - () + 0.00)

checking for cyclic := effects --- OK.

ff: search configuration is  best-first on 1*g(s) + 5*h(s) where
    metric is ((1.00*[RF0](TOTALINTEREST)) - () + 0.00)

advancing to distance:   12
                         11
                         10
                          9
                          8
                          7
                          6
                          5
                          4
                          3
                          2
                          1
                          0

ff: found legal plan as follows

step    0: FLYANDSTAY ORIGIN CITYB HOTELB_A
        1: FLYANDSTAY CITYB CITYC HOTELC_A
        2: FLYANDSTAY CITYC CITYF HOTELF_A
        3: FLYANDSTAY CITYF CITYM HOTELM_A
        4: INCREASESTAY CITYM
        5: INCREASESTAY CITYM
        6: FLYANDSTAY CITYM CITYH HOTELH_A
        7: FLYANDSTAY CITYH CITYD HOTELD_A
        8: FLYANDSTAY CITYD CITYO HOTELO_A
        9: INCREASESTAY CITYO
       10: INCREASESTAY CITYO
       11: FLYANDSTAY CITYO CITYN HOTELN_A
       12: FLYANDSTAY CITYN CITYL HOTELL_A
       13: FLYANDSTAY CITYL CITYK HOTELK_A
       14: INCREASESTAY CITYN
       15: INCREASESTAY CITYN
       16: FLYANDSTAY CITYK CITYI HOTELI_A
       17: FLYANDSTAY CITYI CITYA HOTELA_A
     

time spent:    0.00 seconds instantiating 196 easy, 0 hard action templates
               0.00 seconds reachability analysis, yielding 47 facts and 195 actions
               0.00 seconds creating final representation with 46 relevant facts, 34 relevant fluents
               0.00 seconds computing LNF
               0.00 seconds building connectivity graph
               0.45 seconds searching, evaluating 22282 states, to a max depth of 0
               0.45 seconds total time

